sys/phydat/phydat_str.c

```diff
@@ -64,6 +64,8 @@ const char *phydat_unit_to_str(uint8_t unit)
         case UNIT_TEMP_K:   return "K";
         case UNIT_LUX:      return "lx";
         case UNIT_M:        return "m";
+        case UNIT_M2:       return "sqm";
+        case UNIT_M3:       return "cbm";
```

Comment:
> So we either have to fix that, or can't use UNIT_M2 and UNIT_M3

How about scaling down for these cases:

```C
char phydat_scale_to_str(int8_t scale, uint8_t unit)
{
    switch (unit) {
        case UNIT_M2:
            scale /= 2;
            break;
        case UNIT_M3:
            scale /= 3;
            break;
    }
    switch (scale) {
    ...
```

Due to the API change I don't know however how popular that would be with @haukepetersen. ;-)